standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -5390,6 +5406,493 @@ public void countOpenTxns() throws MetaException {
+            if (errorMsg != null) {
+              throw new MetaException(errorMsg);
+            }
+          }
+
+          if (isAcidTable &&
+                  (!areTxnStatsSupported || !ObjectStore.isCurrentStatsValidForTheQuery(oldParameter, partWriteId,
+                          validWriteIds, true))) {
+            statementDelete.setLong(1, partId);
+            statementDelete.addBatch();
+            numDelete++;
+            if (numDelete == maxNumRows) {
+              statementDelete.executeBatch();
+              numDelete = 0;
+            }
+            LOG.debug("Removed COLUMN_STATS_ACCURATE from the parameters of the partition "
+                    + colStats.getStatsDesc().getDbName() + "." + colStats.getStatsDesc().getTableName() + "."
+                    + colStats.getStatsDesc().getPartName());
+          } else {
+            statementUpdate.setString(1, newParameter.get(COLUMN_STATS_ACCURATE));
+            statementUpdate.setLong(2, partId);
+            statementUpdate.addBatch();
+            numUpdate++;
+            if (numUpdate == maxNumRows) {
+              statementUpdate.executeBatch();
+              numUpdate = 0;
+            }
+            LOG.debug(" Executing update " + statementUpdate);
+          }
+        }
+        result.put((String) entry.getKey(), newParameter);
+      }
+
+      if (numInsert != 0) {
+        statementInsert.executeBatch();
+      }
+
+      if (numUpdate != 0) {
+        statementUpdate.executeBatch();
+      }
+
+      if (numDelete != 0) {
+        statementDelete.executeBatch();
+      }
+
+      if (isAcidTable) {
+        LOG.info("ETL_PERF started updateWriteIdForPartitions ");
+        updateWriteIdForPartitions(dbConn, writeId, partIdList);
+        LOG.info("ETL_PERF done updateWriteIdForPartitions ");
+      }
+      return result;
+    } finally {
+      closeStmt(stmtInt);
+      closeStmt(statementInsert);
+      closeStmt(statementUpdate);
+      closeStmt(statementDelete);
+    }
+  }
+
+  private static class PartitionInfo {
+    long partitionId;
+    long writeId;
+    public PartitionInfo(long partitionId, long writeId) {
+      this.partitionId = partitionId;
+      this.writeId = writeId;
+    }
+  }
+
+  private Map<String, PartitionInfo> getPartitionInfo(Connection dbConn, String catName,
+                                                      String dbName, String tblName,
+                                                      List<String> partKeys) throws SQLException {
+    List<String> queries = new ArrayList<>();
+    StringBuilder prefix = new StringBuilder();
+    StringBuilder suffix = new StringBuilder();
+    PreparedStatement pStmt = null;
+    ResultSet rs = null;
+    Map<String, PartitionInfo> partitionInfoMap = new HashMap<>();
+
+    try {
+      long tblId;
+      Statement stmt = dbConn.createStatement();
+      rs = stmt.executeQuery("select \"TBL_ID\" from \"DBS\", \"TBLS\" where \"DBS\".\"NAME\" = "
```

Comment:
Try to move SQL into the constants with parametrization, see MessageFormat.format

Reply:
done